rllib/offline/estimators/feature_importance.py

```diff
@@ -0,0 +1,64 @@
+# TODO (@Kourosh) move this to a better location and consolidate the parent class with
+# OPE
+
+from typing import Callable, List
+from ray.rllib.policy import Policy
+from ray.rllib.utils.typing import SampleBatchType
+from ray.rllib.offline.estimators.off_policy_estimator import (
+    OffPolicyEstimator,
+    OffPolicyEstimate,
+)
+
+import numpy as np
+import copy
+
+
+def perturb_fn(batch: np.ndarray, index: int):
+    # shuffle the indexth column features
+    random_inds = np.random.permutation(batch.shape[0])
+    batch[:, index] = batch[random_inds, index]
+
+
+class FeatureImportance(OffPolicyEstimator):
+    def __init__(
+        self,
+        name: str,
+        policy: Policy,
+        gamma: float,
+        repeat: int = 1,
+        perturb_fn: Callable[[np.ndarray, int], None] = perturb_fn,
+    ):
+        """
+        Args:
+            name: string to save the feature importance results under.
+            policy: the policy to use for feature importance.
+            repeat: number of times to repeat the perturbation.
+            gamma: dummy discount factor to be passed to the super class.
+            perturb_fn: function to perturb the features. By default reshuffle the
+            features within the batch.
+        """
+        super().__init__(name, policy, gamma=gamma)
+        self.repeat = repeat
+        self.perturb_fn = perturb_fn
+
+    def estimate(self, batch: SampleBatchType) -> List[OffPolicyEstimate]:
```

Comment:
stupid question, where are these metrics averaged across batches?

Reply:
here: https://github.com/kouroshHakha/ray/blob/master/rllib/evaluation/rollout_worker.py#L1187
it's part of how OffPolicyEstimators are integrated into RLlib.